scripts/hid_configurator/configurator_core.py

```diff
@@ -208,6 +248,45 @@ def __str__(self):
                 '  Version: {}.{}.{}.{}').format(self.flash_area_id, self.image_len, self.ver_major, self.ver_minor,
                                                  self.ver_rev, self.ver_build_nr)
 
+class ConfigParser:
+    """ Class used to simplify "read-modify-write" handling of parameter structs.
+        For example when the python argument modifies one value within struct,
+        and the remaining struct values should remain unchanged. """
+
+    def __init__(self, fetched_data, fmt, member_names, value_presenter, value_formatter):
+        assert struct.calcsize(fmt) <= EVENT_DATA_LEN_MAX
+        self.fmt = fmt
+
+        if value_presenter is not None:
+            self.presenter = value_presenter
+        else:
+            self.presenter = lambda x: x
+
+        if value_formatter is not None:
+            self.formatter = value_formatter
+        else:
+            self.formatter = lambda x: x
+
+        vals = struct.unpack(self.fmt, fetched_data)
+        if len(member_names) != len(vals):
+            raise ValueError('format does not match member name list')
+        self.members = collections.OrderedDict()
+        for key, val in zip(member_names, vals):
+            self.members[key] = val
+
+    def __str__(self):
+        return ''.join(map(lambda x: '{}: {}\n'.format(x[0], x[1]), self.members.items()))
+
+    def config_get(self, name):
+        return self.presenter(self.members[name])
+
+    def config_update(self, name, val):
+        if name not in self.members:
```

Comment:
Do we need this? Isn't `KeyError` raised automatically if we try to use key which is not in the dictionary ?

Reply:
If the key is not already there it will just be created with the new value, which is not the intention of this function. There might be a more elegant way to prevent new keys from being added after the initial creation that I'm not aware of though